webapp/builder/views.py

```diff
@@ -8,4 +10,40 @@
 
 @builder.route("/")
 def index():
-    return flask.render_template("builder/index.html"), 200
+    return flask.render_template("builder/index.html")
+
+
+@builder.route("/auth/authenticate")
+def authenticate():
+    secret = logic.set_auth_secret(flask.session)
+    redirect_url = logic.get_oauth_url(flask.request, state=secret)
+    return flask.redirect(redirect_url)
+
+
+@builder.route("/auth/verify", methods=["GET", "POST"])
+def verify():
+    secret = flask.request.args.get("state")
+    if not logic.valid_auth_secret(flask.session, secret):
+        # Secret does not match our copy warn about potential CSRF attack
+        raise GithubAuthException(
+            "Returned shared secret does not match generated shared secret"
+        )
+
+    code = flask.request.args.get("code")
+    response = logic.exchange_code_for_token(flask.request, code)
+
+    data = response.json()
```

Comment:
it already happened that we got an invalid json. You should surround it with try catch ValueError (if I remember well)

Reply:
JsonDecodeError?